core/client/fs/src/main/java/alluxio/client/file/RetryHandlingFileSystemMasterClient.java

```diff
@@ -234,6 +238,21 @@ public long getNewBlockIdForFile(final AlluxioURI path)
     }, RPC_LOG, "ListStatus", "path=%s,options=%s", path, options);
   }
 
+  @Override
+  public Stream<URIStatus> listStatusInStream(
+      final AlluxioURI path, final ListStatusPOptions options)
+      throws AlluxioStatusException {
+    return retryRPC(() ->
+      StreamSupport.stream(Spliterators.spliteratorUnknownSize(
+          mClient.listStatus(ListStatusPRequest.newBuilder().setPath(getTransportPath(path))
```

Comment:
Does `listStatus` just return the entire array/list? How does this stream anything when the result is already just a list/array?

Reply:
`listStatus` returns an iterator where each element of this iterator is a list. This is the recommended way to implement streaming server gRPC calls. e.g., the RPC returns  something like `[a,b, ...], [c,d, ...]` depending on the batch size. 

Note that, in the original implementation of client-side listStatus 
https://github.com/Alluxio/alluxio/pull/12315/files#diff-f0b3f5c51c2e3e48833beccd832c5a63de3d74b8d195d9da564c99c299f50288R229-R237

We create an extra list and add each element in [a,b,...]  and [c,d,...] to this list before return to cli, which increased the resource overhead (may not be big) but also latency.

@gpang

Reply:
In other words, Alluxio master is already streaming results to client. however, we are not taking advantage of this streaming API given that in `RetryHandlingFileSystemMasterClient.java`, we will simply wait until all results are streamed from master before returning to client API callers, causing overhead in resource (e.g., memory pressure) and latency. Now we directly wrap the stream of status (may still in transmission) to client so they are able to proceed with partially transmitted results.

Reply:
Ahhh, I see. I didn't realize that `listStatus` returned an `Iterator`.